### DISCUSSION TITLE
Problem with setting the default value for a ManyToOne relation but only when used in livecomponent with livecollection

### DISCUSSION THREAD

**Discussion by gewait** [symfony/ux#2241] [Questions & Answers]
I have an entity "Main" which has a OneToMany relation named "children" to the entity "Child".

    class Main
    {
        #[ORM\OneToMany(targetEntity: Child::class, mappedBy: 'main')]
        private Collection $children;

----

Inside the "Child" there is a ManyToOne relation named "other" to another entity named "Related".

    class Child
    {
        #[ORM\ManyToOne]
        private ?Related $other = null;

----

When a new child is created I want to set a default value for the relation "other".

I can't do it in the constructor of the entity because I would need the entitymanager for that.

I can't use a factory class, because I want to use this in a livecomponent with a livecollection 
and there the creation of the objects is done internally.

So I do it via the data option in the formtype for child:

        src/Form/ChildType.php

        $builder
            ->add('other', EntityType::class, [
                'required' => false,
                'class' => Related::class,
                'choice_label' => 'name',
                'data' => $this->relatedRepository->findOneByName('when_this_is_preselected_then_it_works'),  //set default
            ])

----

This *does work* in a normal form

but does *not work* in a *live component* with a *live collection*.

----

Since the live collection does all the work internally when creating/adding a new child, 
I don't know how or where I can initialize the child.other relation property!

----

Is this the expected behaviour? What am I doing wrong? Can someone help?

----

Reproducer:

[live-collection-reproducer-zip.zip](https://github.com/user-attachments/files/17191140/live-collection-reproducer-zip.zip)

**Comment by smnandre** [comment]
Do you already made it work with a standard CollectionType ?

**Reply by gewait** [reply]
No, because some weeks ago I was trying to use a standard CollectionType with the option "allow_add" inside a livecomponent, which did not work (see https://github.com/symfony/ux/issues/1973).

Since then I was under the impression that I have to use the livecollectiontype in livecomponents when using the option "allow_add".

**Reply by gewait** [reply]
Friendly ping :-) Do you need more information or can you give me a hint on what I can do?

**Comment by tsiatka** [comment]
same issue here, the data option is not working when using the addCollectionItem

**Comment by imbachb** [comment]
Had the same issue. In the symfony slack channel someone suggested to override the `addCollectionItem` Method from `LiveCollectionTrait` and set the default values accordingly, i.e. change the line where the values are set from
```
$propertyAccessor->setValue($this->formValues, $propertyPath."[$index]", []);
```
to
```
$propertyAccessor->setValue($this->formValues, $propertyPath."[$index]", ['someProperty' => 'defaultValue']);
```
A bit cumbersome to have to override this function but it does work.
Maybe there is a way to access the form type's `data` field here so we would only have to define the default values in the `Form/SomeType.php` class.
